Rust/src/common/directionaldensity.rs

```diff
@@ -0,0 +1,128 @@
+
+use crate::common::divector::DiVector;
+use crate::samplerplustree::boundingbox::BoundingBox;
+
+#[repr(C)]
+#[derive(Clone)]
+pub struct InterpolationMeasure {
+    pub measure: DiVector,
+    pub distance: DiVector,
+    pub probability_mass: DiVector,
+    pub sample_size: f32
+}
+
+impl InterpolationMeasure {
+    pub fn empty(dimension: usize, sample_size: f32)  -> Self {
+        InterpolationMeasure {
+            measure: DiVector::empty(dimension),
+            distance: DiVector::empty(dimension),
+            probability_mass: DiVector::empty(dimension),
+            sample_size,
+        }
+    }
+
+    pub fn new(measure : DiVector, distance: DiVector, prob_mass:DiVector,sample_size: f32) -> Self{
+        assert!(measure.dimension() == distance.dimension(), " incorrect lengths");
+        assert!(measure.dimension() == prob_mass.dimension(), " incorrect lengths");
+        InterpolationMeasure{
+            measure: measure.clone(),
```

Comment:
Do we need to call `clone` here? It looks like the function takes ownership of the arguments.

Reply:
No., not needed. Fixed.